pkg/cmd/stack/deploy.go

```diff
@@ -15,115 +15,170 @@ package stack
-			if r.Name != stackHelmRepoName {
-				continue
-			}
-			re = r
-			break
-		}
-	}
-	if re == nil {
-		if err := helm.AddRepo(settings, stackHelmRepoName, stackHelmRepoURL, stackHelmChartName, stackHelmChartVersion); err != nil {
-			return err
-		}
-		log.Information("'%s' has been added to your helm repositories.", stackHelmRepoName)
-	} else {
-		err := helm.UpdateRepo(re, settings, stackHelmChartName)
-		if err != nil {
-			return err
-		}
+	if err := configmaps.Deploy(ctx, cfg, s.Namespace, c); err != nil {
+		return err
 	}
 
+	return err
+}
+
+func deploy(ctx context.Context, s *model.Stack, forceBuild, wait, noCache bool, c *kubernetes.Clientset) error {
 	spinner := utils.NewSpinner(fmt.Sprintf("Deploying stack '%s'...", s.Name))
 	spinner.Start()
 	defer spinner.Stop()
 
-	actionConfig := new(action.Configuration)
-	if err := actionConfig.Init(settings.RESTClientGetter(), s.Namespace, helmDriver, func(format string, v ...interface{}) {
-		message := strings.TrimSuffix(fmt.Sprintf(format, v...), "\n")
-		spinner.Update(fmt.Sprintf("%s...", message))
-	}); err != nil {
-		return fmt.Errorf("error initializing stack client: %s", err)
+	if err := translate(ctx, s, forceBuild, noCache); err != nil {
+		return err
 	}
 
-	exists, err := helm.ReleaseExist(action.NewList(actionConfig), s.Name)
-	if err != nil {
-		return fmt.Errorf("error listing stacks: %s", err)
+	for name := range s.Services {
+		if len(s.Services[name].Volumes) == 0 {
+			if err := deployDeployment(ctx, name, s, c); err != nil {
+				return err
+			}
+		} else {
+			if err := deployStatefulSet(ctx, name, s, c); err != nil {
+				return err
+			}
+		}
+		if len(s.Services[name].Ports) > 0 {
+			svcK8s := translateService(name, s)
+			if err := services.Create(ctx, svcK8s, c); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := destroyServicesNotInStack(ctx, s, c); err != nil {
+		return err
 	}
-	if exists {
-		return helm.Upgrade(action.NewUpgrade(actionConfig), settings, s, stackHelmRepoName, stackHelmChartName, stackHelmChartVersion, vals, wait)
+
+	if !wait {
+		return nil
 	}
-	return helm.Install(action.NewInstall(actionConfig), settings, s, stackHelmRepoName, stackHelmChartName, stackHelmChartVersion, vals, wait)
-}
 
-func isNotExist(err error) bool {
-	return os.IsNotExist(errors.Cause(err))
+	spinner.Update("Waiting for services to be ready...")
+	return waitForPodsToBeRunning(ctx, s, c)
+
 }
 
-func saveStackFile(s *model.Stack) (string, error) {
-	tmpFile, err := ioutil.TempFile("", "okteto-stack")
-	if err != nil {
-		return "", fmt.Errorf("failed to create dynamic stack manifest file: %s", err)
+func deployDeployment(ctx context.Context, svcName string, s *model.Stack, c *kubernetes.Clientset) error {
+	d := translateDeployment(svcName, s)
+	old, err := c.AppsV1().Deployments(s.Namespace).Get(ctx, svcName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error getting kubernetes deployment: %s", err)
```

Comment:
I think this is the kind of error we shouldn't get back to the user, since it breaks the abstraction.  could we change it to be more specific to the context/terminology we use in stack?